File: INFO3410_A2/test/testMovieGenre.py

# # darkWorld = Movie()
# # darkWorld.title = "Dark World"
# # print(thor.addRelatedMovie(darkWorld))
# # print(thor.related[0].title)


from movieGenre import *

Batman = Movie()
Batman.uuid = '01'
Batman.title = 'dark night'
Batman.year = 2010


superman = Movie()
superman.uuid = 'S01'
superman. title = 'Superman'
superman.year = 2015


hero = Genre()
hero.name = 'hero'

action = Genre()
action.name = 'Action'

fantasy = Genre()
fantasy.name = 'Fantasy'


print('Add Related Movie')
def test_addRelated(a, b):
    if (a.addRelatedMovie(b) == True):
        print('true')
    else:
        print('false')
test_addRelated(Batman, superman); 



print('\n setGenre ')
def test_setGenre(m, g):
    if (m.setGenre(g) == True):
        m.genres = g.name
        print('True')
    else:
        print('TEST FAILED')
test_setGenre(Batman, action); # test case
test_setGenre(superman, hero); 


# addMovie Test Cases
print('\n addMovie ')
def test_addMovie(g, m):
    if (g.addMovie(m) == True):
        print('True')
    else:
        print('False')
test_addMovie(action, Batman); # test case
test_addMovie(hero, superman); 


# display movies
# print('\n', thor)
# print('\n', avengers)
# print('\n', spiderman)

# display genres
# print('\n', fantasy)
# print('\n', hero)
# print('\n', action)
        

File: INFO3410_A2/test/testMovieGenre.js
// module = require("./movieGenre")
// let movie = Movie();
// movie = Movie()
// darkWorld = new Movie()
// thor = new Movie()
// action = new Genre()
// action.name = "Action"
// darkWorld.title = "Dark World"
// thor.title = "Thor"
// thor.genres = action.name
// darkWorld.genres = action.name
// darkWorld.addRelatedMovie(thor);
// thor.addRelatedMovie(darkWorld);

// console.log(thor.related[0].title)
// console.log(thor.genres)

//const Movie = require('./movieGenre').Movie;
//const Genre = require('./movieGenre').Genre;
//const assert = require('assert');

//module = require('./movieGenre.js')


 let movie = require("./movieGenre.js");
 let Movie = movie.Movie;
 let genre = require("./movieGenre.js");
 let Genre = genre.Genre;





batman = new Movie();
batman.uuid = '1';
batman.title = 'Batman';
batman.year = 2010;

superman = new Movie();
superman.uuid = '2';
superman. title = 'Superman';
superman.year = 2011;



console.log(' addRelatedMovie Test Cases');
function test_addRelated(a, b){
    
    if (a.addRelatedMovie(b) === true){
    
        console.log('TEST PASSED');
    } else {
        console.log('TEST FAILED');
    }
}
test_addRelated(batman, superman); 
